samples/popup.html

```diff
@@ -0,0 +1,204 @@
+      ></a-box>
+      <a-entity
+        id="primaryCamera"
+        position="0 0 9"
+        rotation="0 0 0"
+        camera
+        look-controls
+        wasd-controls
+        kinematic-body
+      ></a-entity>
+      <a-entity id="readRoom" read-room position="0 -100 0">
+        <a-entity
+          id="blackboard"
+          info-panel
+          visible="true"
+          geometry="primitive: plane; width: 18   ; height: 15"
+          material="color: #333333; shader: flat; transparent: false"
+        >
+        </a-entity>
+        <a-entity id="returnButton"></a-entity>
+        <a-entity
+          id="popup-camera"
+          position="0 0 9"
+          rotation="0 0 0"
+          camera="active: false;"
+        ></a-entity>
+      </a-entity>
+    </a-scene>
+  </body>
+</html>
+<script>
+  var jsonDataArr = [
+    'jsonData: {"width": "1", "height": "2", "depth": "1", "color": "orange", "x": "2", "y": "0", "z": "0"}',
+    'jsonData: {"width": "1", "height": "2", "depth": "1", "color": "blue", "x": "0", "y": "0", "z": "0"}',
+  ];
+
+  setupEnv(jsonDataArr);
+
+  function setupEnv(jsonDataArr) {
+    // default return button. Always here.
+    // TODO: move into a A-Frame component. [closePopup]
+    entityEl = document.querySelector("#returnButton");
+    entityEl.setAttribute(
+      "chair",
+      'jsonData: {"width": "1", "height": "1", "depth": "0.01", "color": "red", "x": "-8.25", "y": "6.75", "z": "0.005"}'
+    );
+    entityEl.addEventListener("click", function () {
+      var primaryCamera = document.querySelector("#primaryCamera");
+      primaryCamera.setAttribute("camera", "active", true);
+      wipeBlackboard();
+    });
+
+    // second button. May or may not be here. There can be many of these.
+    // TODO: Also move into a component. [openPopup]
+    var parentEl = document.querySelector("a-scene");
+    var entityEl2 = document.createElement("a-entity");
+    entityEl2.setAttribute("chair", jsonDataArr[1]);
+    entityEl2.addEventListener("click", function () {
+      addEntityToBlackboard(
+        '{"componentType": "chair", "jsonData": {"width": "10", "height": "2", "depth": "1", "color": "orange", "x": "2", "y": "0", "z": "2"}}'
+      );
+      var popupCameraEl = document.querySelector("#popup-camera");
+      popupCameraEl.setAttribute("camera", "active", true);
+    });
+    parentEl.appendChild(entityEl2);
+  }
+
+  // TODO: Add support for multiple entity creation
+  function addEntityToBlackboard(componentData) {
+    var componentDataParsed = JSON.parse(componentData);
+    var blackboardEl = document.querySelector("#blackboard");
+    var entityEl = document.createElement("a-entity");
+    entityEl.setAttribute(
+      componentDataParsed.componentType,
+      "jsonData",
+      JSON.stringify(componentDataParsed.jsonData)
+    );
+    blackboardEl.appendChild(entityEl);
```

Comment:
I think `appendChild` is an async function. So before switching the camera, I believe we should listen for the `loaded` event: https://aframe.io/docs/1.2.0/introduction/javascript-events-dom-apis.html#adding-an-entity-with-appendchild

Reply:
I added a listener although I don't think it is having any impact on it in our case. I might be mistaken.

Reply:
Yeah, for now since we have basic rendering it won't affect it much, but it will affect it if we have more complex objects being rendered :) Like puzzles